generators/base/generator-base.mjs

```diff
@@ -255,11 +253,7 @@ export default class JHipsterBaseGenerator extends PrivateBase {
    * @return {boolean} true if the User is built-in.
    */
   isUsingBuiltInUser() {
```

Comment:
It's used only at https://github.com/jhipster/generator-jhipster/blob/c4223686a9457e99c50a41584b950f3cafa8e23c/generators/server/generator.mjs#L302.
Remove this api and move to server generator.

Reply:
Not sure that it's server only: there may be some impact on the client too (for example generating the registration form)

Reply:
Maybe you are confusing the derived property `application.builtInUser` with the application implementation.
Look close at the logic:
```
 return (
     !config ||
     (!config.skipUserManagement && config.databaseType !== NO_DATABASE) ||
     (config.authenticationType === OAUTH2 && config.databaseType !== NO_DATABASE)
   );
```

The server defines the `application.builtInUser` value and client generator uses it.
So in this case the logic is server only, and the client uses it.
Our client definition just mirrors what the server implements.

Reply:
No confusing with derived props, just challenging why this derived property is related to `server`. To me it should be an `entities` global logic and derived property assignment should be there.

Reply:
We can discuss it in front of a virtual beer ^^. Microservice+Jwt and Gw+Oauth is not a `server` concern, but an `application` one.
I may be confused, but to me `server` is related to one service's backend, for example the backend of one microservice or the backend of one monolith, which is why I didn't want to put that method here.
Am I wrong?

As an illustration: I've no use case in mind where the `isBuiltInUser` method will return `true`at the `gateway`level while it will return `false`at the `microservice` level within the same `application`.